### ISSUE TITLE
Animated tile IDs, and nullptr returned by tileset->getTile(tileId), local vs global

### ISSUE THREAD

**Issue by Moros1138** [SSBMTonberry/tileson#110]
https://github.com/SSBMTonberry/tileson/blob/4a9d556ab3c8b2f1a4197a0339c4498560a6fdc8/examples/sfml/SfmlDemoManager.cpp#L304

This fails to account for the idea of multiple tilesets in a single map. I just spent the last 3 hours (or so) trying to figure out why my ``animatedTile`` was always ``nullptr``

To save others from the same frustration here's what works for me. I had to modify the animation code in my own project in this way, but I figured I'd use the examples because it's how people are learning to use Tileson.

```cpp
            uint32_t ownerId = tileObject.getTile()->getId() + tileset->getFirstgid() - 1;  // HERE
            if(m_animationUpdateQueue.count(ownerId) == 0) //This is only built once to track all tile IDs with animations
                m_animationUpdateQueue[ownerId] = &tileObject.getTile()->getAnimation();

            uint32_t tileId = tileObject.getTile()->getAnimation().getCurrentTileId() + tileset->getFirstgid() -1; // AND HERE...
            tson::Tile *animatedTile = tileset->getTile(tileId);
            drawingRect = animatedTile->getDrawingRect();
```

I don't know that this is indicative of a problem with Tileson and the way it handles Ids, but I wanted to share this so hopefully others don't befall my fate.

**Comment by SSBMTonberry** [comment]
Thank you for posting this, @Moros1138 :slightly_smiling_face: 
I recon the animation part of the examples is pretty old, so there might be some things it does not consider, and should be taken with a grain of salt. At the same time, this seems like an issue that `Tileson` **should** be able to solve for you, or at least make more easy or straight forward to resolve. I'll put this into the current roadmap. I'm sure something can be done to make this more intuitive and easy to use! The idea behind a lot of the features exclusive to `Tileson` is to make it as easy as possible to use a `Tiled` map without having to deal with unnecessary manual mappings that can be done beforehand :smile:

**Comment by Moros1138** [comment]
First, you're very welcome. Regarding the bug label. I, myself, wasn't particularly comfortable with calling it a bug when I was crafting the issue/post. I opted to approach it from the "it might save someone else some hassle" kinda post!

If I might briefly indulge in non sequitur. It's been great being able to integrate Tiled maps into my own projects. Originally I rawdogged JSON parsing and doing it all myself. Then I found out you already suffered for me! Thanks for Tileson!
